packages/react-native/React/Views/RCTViewUtils.m

```diff
@@ -20,3 +20,63 @@ UIEdgeInsets RCTContentInsets(UIView *view)
   }
   return UIEdgeInsetsZero;
 }
+
+
+// https://drafts.csswg.org/css-transforms/#transform-origin-property
+void applyTransformWithTransformOrigin(UIView* view, NSString *transformOrigin, CATransform3D transform) {
```

Comment:
In this function, we could process top, left etc enums on JS side and pass percentages to the native. It can save us from maintaining two implementations. let me know!